src/test/crimson/seastore/transaction_manager_test_state.h

```diff
@@ -223,10 +262,31 @@ class TMTestState : public EphemeralTestState {
       return static_cast<journal::CircularBoundedJournal*>(tm->get_journal())->mkfs(
 	config
       ).safe_then([this]() {
-	return tm->mkfs(
-	).handle_error(
-	  crimson::ct_error::assert_all{"Error in mkfs"}
-	);
+	return seastar::do_with(
+	  static_cast<RBMCleaner*>(epm->get_cleaner())->get_rb_group()->get_rb_managers(),
+	  [this](auto &rb_managers) {
+	  return crimson::do_for_each(
+	    rb_managers.begin(),
+	    rb_managers.end(),
+	    [](auto &rb)
+	  {
+	    RandomBlockManager::mkfs_config_t rbm_config;
+	    random_block_device::RBMDevice* dev =
+	      static_cast<random_block_device::RBMDevice*>(rb->get_device());
+	    device_id_t d_id = dev->get_device_id();
+	    rbm_config.start = paddr_t::make_blk_paddr(d_id, 0);
+	    rbm_config.end = paddr_t::make_blk_paddr(d_id, dev->get_size());
+	    rbm_config.block_size = dev->get_block_size();
+	    rbm_config.total_size = dev->get_size();
+	    rbm_config.device_id = d_id;
+	    return rb->mkfs(rbm_config);
```

Comment:
Moved to RBMDeviceGroup. Can you re-check that the change is the right way?

Reply:
Not really. I'm not convinced that the mkfs path for RBM devices needs to start at `RandomBlockManager::mkfs(mkfs_config_t)` instead of `Device::mkfs(device_config_t)`.

The observation is that `NVMeBlockDevice::mkfs(device_config_t)` is an empty method, making the original mkfs path from `SeaStore` to `Device` completely ignored (i.e. the "the exisiting paths" I mentioned above). The original path was designed to all kinds of devices, so that the high-level mkfs logic can be resued, with the specializations encapsulated below the `Device` interface.

Also refer to how `ZNSSegmentManager::mkfs()` and `BlockSegmentManager::mkfs()` were specialized for different devices.

Reply:
To solve this, I think current RandomBlockManager needs to be inherited from Device class. I added a commit to see if my thought is correct---this will be re-arranged.

Reply:
For segmented implementations, the counterpart of `RandomBlockManager` is the `SegmentCleaner`, which implements space tracking (mark_space_free/used) and the segment allocator. It means that originally, the `Device` abstraction was introduced to represent low-level device managements.

If there is no strong reason to move/merge `RandomBlockManager` with `Device`, I intend to think move/merge `RandomBlockManager` with `RBMCleaner` is more appropriate, matching the existing abstractions better.

Reply:
I moved mkfs to RBMDevice based on your comment (see the last commit---this is just for discussion). If this approach makes sense to you,  I think we can implement RBM related cleanup (remove bitmap layout and moving mkfs to RBMDevice) first as a separate PR before continuing other works. What do you think?

Reply:
Ok. I'll post a PR.